internal/hcl/evaluator.go

```diff
@@ -107,6 +107,7 @@ func (e *Evaluator) Run() ([]*Module, error) {
 
 	// let's load the modules now we have our top level context.
 	e.moduleCalls = e.loadModules()
+	e.evaluate(lastContext)
```

Comment:
evaluate the context again before expanding the blocks. So that module outputs are available for expressions.